pennylane/transforms/optimization/remove_barrier.py

```diff
@@ -0,0 +1,76 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Transform for remove Barrier gate in quantum circuits."""
+# pylint: disable=too-many-branches
+from pennylane import apply
+from pennylane.transforms import qfunc_transform
+
+
+@qfunc_transform
+def remove_barrier(tape):
+    """Quantum function transform to remove Barrier gate.
+
+    Args:
+        qfunc (function): A quantum function.
```

Comment:
Out of scope: it seems weird to have `qfunc` here but `tape` in the signature (even if we decorate with `qfunc_transform`) :thinking:

Reply:
I followed the structure of the other transformers that pass qfunc as an argument.